packages/angular_devkit/build_angular/src/angular-cli-files/utilities/bundle-calculator.ts

```diff
@@ -108,11 +266,11 @@ class AllCalculator extends Calculator {
  */
 class AnyComponentStyleCalculator extends Calculator {
   calculate() {
-    return Object.keys(this.compilation.assets)
-      .filter(key => key.endsWith('.css'))
-      .map(key => ({
-        size: this.compilation.assets[key].size(),
-        label: key,
+    return this.assets
```

Comment:
This line here doesn’t seem right, the css assets here would be global css file and not the component styles.

Reply:
Ah, yes I think you're right. Is there a way to get only component styles from Webpack stats? I don't see component-specific styles in those statistics. Is there some way I can query or search all component styles specifically?

Reply:
Did this work previously?

This appears to be fairly complicated to do correctly since the process will differ greatly between JIT and AOT modes.  For JIT, this would need to hook into the ngtools JIT decorator transformer to determine the styles for each component and then calculate the sizes of any file based stylesheets.  For AOT, the resource loader runs a child webpack compilation which only outputs the resource (in this case the stylesheet) for the component.  As long as this plugin is provided to the child webpack compilation, this should work as is.  
However, in either case, this calculator should not be present in the main webpack compilation as none of the assets will be component stylesheets.

Reply:
If this aspect is working as it did before this PR, I think we should defer any behavior changes here to a separate PR.

Reply:
I just tested this and component styles budget checks are no longer working. For differential builds, the plugin is disabled, so we must compute this after the build. Unfortunately we no longer have access the `compilation` property, so I don't think we can easily compute this in the old way.

Regarding JIT vs AOT, these metrics are only output if the user runs with `--prod` I believe. Is JIT still used in that context? Would we be able to get away with just getting AOT to work?

What would be the best way to get access to the underlying CSS file size data? Would we need to make another plugin that just handles CSS styles, or run the existing plugin when building differential builds, but only compute style data?

Reply:
JIT in production is rare and it doesn't appear that it worked previously either.
For AOT, the plugin could be added to the child compilation directly ([code](https://github.com/angular/angular-cli/blob/master/packages/ngtools/webpack/src/resource_loader.ts#L64)).  The options would need to be available at that location though.  That will probably need some thought as ideally the budget options should probably not be passed through the ngtools webpack plugin.

Reply:
After some experimentation, it seems that component style budgets need to be checked in a Webpack plugin so their size information can be retrieved while we are still able to emit usable errors.

There is an existing plugin, but that is only used to check budgets for non-differential builds. I pulled out the `AnyComponentStyle` functionality to a separate plugin which is used for all builds.